File: Week 1/MiniMaxSum.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.List;
import java.util.stream.Stream;
import java.util.stream.Collectors;


public class MiniMaxSum {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        List<Integer> arr = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(Collectors.toList());

        Result.miniMaxSum(arr);

        bufferedReader.close();
    }
    
  class Result{

    public static void miniMaxSum(List<Integer> arr) {
            long min = Long.MAX_VALUE;
            long max = Long.MIN_VALUE;
            long sum = 0;
            
            for (int i = 0; i < arr.size(); i++) {
                sum += arr.get(i);
                min = Math.min(min, arr.get(i));
                max = Math.max(max, arr.get(i));
            }
                
        // Calculate the minimum and maximum sums
        long minSum = sum - max; // Subtracting the largest element
        long maxSum = sum - min; // Subtracting the smallest element
        
        System.out.println(minSum + " " + maxSum);
    }
  }
}

File: Week 1/MiniMaxSum.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'miniMaxSum' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function miniMaxSum(arr) {
    // Write your code here
   let min  = Math.min(...arr);
   let max = Math.max(...arr);
   let sum = 0;
   
   for(let i = 0; i < arr.length; i++){
        sum += arr[i];
        min = Math.min(min, arr[i]);
        max = Math.max(max, arr[i]);
   }
    console.log((sum - max) + " " + (sum - min));
}

function main() {

    const arr = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));

    miniMaxSum(arr);
}
